build/libraw.bat

```diff
@@ -1,12 +1,30 @@
 @echo off
 title Building LibRaw
 set arch=%1
-for /f "usebackq tokens=1* delims=" %%x in (`vswhere -find **\vcvarsall.bat`) do set vcvar="%%~x"
+set vcvar=
 
-if not defined vcvar (
-	echo "unable to find 'vcvarsall.bat'. visual studio c++ package needs to be installed"
-	exit /b 2
-)
+set progFiles="%ProgramFiles%\Microsoft Visual Studio\Installer\vswhere.exe"
+for /f "usebackq tokens=1 delims=" %%x in (`"%progFiles%" -find **\vcvarsall.bat`) do set vcvar="%%~x"
+
+if defined vcvar goto fileFound
+
+
+if not defined ProgramFiles(x86) goto fileNotFound
+
+
+for /f "usebackq tokens=1 delims=" %%y in (`"%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe" -find **\vcvarsall.bat`) do set vcvar="%%~y"
+if defined vcvar goto fileFound
```

Comment:
"The system cannot find the path specified." was from vswhere when it cant find the file in the x64 directory. I just tried to look for a way to skip the message but Im not seeing a way to turn it off. I could add messages for where it is looking, would at least explain the message.

also, are you able to get it to build properly yet? I think I was still having issues yesterday before I switched to the bug task.

Reply:
I did some digging and have some thoughts on things we can try to fix this. It appears the `for` statement doesn't allow us to directly write to `>nul` or variations of that. But we can pipe the output right into a variable and then check the variable in a similar fashion. 

Could we try and set 2 vars and check the output of them?
```
set progFiles="%ProgramFiles%\Microsoft Visual Studio\Installer\vswhere.exe"
set progFilesX86="%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe"
```

Then we can run the following commands to set the response of `vswhere.exe`
```
set vcvar=%progFiles% -find **\vcvarsall.bat
```
We can then check the output of `vcvar` by using `%vcvar%` which may read "The system cannot find the path specified.". If that is true then we fall down to the next statement

```
set vcvar=%progFilesX86% -find **\vcvarsall.bat
```

Now when we check the value of `vcvar` it'll have the correct locations loaded. 

### Interesting Findings
While I did a little research on this I noticed that `vswhere` grabbed the incorrect value for me. It has 2 entries loaded as I have multiple instances of Visual Studio installed and has VS 2022 at the top of the list. Since we are only supporting VS 2019 at this point we may want to do a hard check for 2019 results
```
D:\>%vcvar%
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\vcvarsall.bat
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvarsall.bat
```

```

Reply:
it actually sets both? pretty sure I set it up so only one would be grabbed

Reply:
latest commit seems to work and avoids weird error messages. but i do not like the code itself. i started to add a lot of goto's so things would actually set correctly. batch files are weird.